src/particles/PhysicsParticles.hpp

```diff
@@ -323,6 +424,116 @@ template <typename ContainerType, ParticleType particleType> class PhysicsPartic
 		}
 	}
 
+	// Implementation of CIC particle creation
+	void createCICParticles(amrex::MultiFab &state, int lev, amrex::Real current_time, amrex::Real dt) const override
+	{
+		if (container_ != nullptr) {
+			const int mass_idx = this->getMassIndex();
+			if (mass_idx >= 0 && mass_idx + 3 < ContainerType::ParticleType::NReal) {
+				const double particle_creation_time_1 = this->getParam1();
+				const double particle_creation_time_2 = this->getParam2();
+
+				for (amrex::MFIter mfi = container_->MakeMFIter(lev); mfi.isValid(); ++mfi) {
+					// TODO(cch): I don't know whether this is needed. WarpX has this.
+					amrex::Gpu::synchronize();
+
+					const auto &box = mfi.validbox();
+					const auto &state_arr = state.array(mfi);
+					const auto &geom = container_->Geom(lev);
+					const auto dx = geom.CellSizeArray();
+					const auto plo = geom.ProbLoArray();
+
+					// Count particles to be created in this box
+					amrex::Gpu::DeviceVector<amrex::Long> counts(box.numPts(), 0); // 1 if cell creates particle, 0 if not
```

Comment:
I think the type of this can be `amrex::Gpu::DeviceVector<unsigned int>` instead.

Reply:
Also, assigning a default value is unnecessary, since you explicitly assign `0` if no particle should be created in the kernel below.